Corona-Warn-App/src/main/java/de/rki/coronawarnapp/coronatest/qrcode/RapidAntigenQrCodeExtractor.kt

```diff
@@ -113,6 +131,16 @@ class RapidAntigenQrCodeExtractor @Inject constructor() : QrCodeExtractor<Corona
             val complete = allOrNothing.all { it } || allOrNothing.all { !it }
             if (!complete) throw InvalidQRCodeException("QRCode contains incomplete personal data: $raw")
         }
+
+        private fun requireValidHash() {
+            val isQrCodeWithPersonalData = firstName != null && lastName != null && dateOfBirth != null
+            val generatedHash =
+                "${raw.dateOfBirth}#${raw.firstName}#${raw.lastName}#${raw.timestamp}#${raw.testid}#${raw.salt}"
+                    .toSHA256()
+            if (isQrCodeWithPersonalData && !generatedHash.equals(hash, true)) {
+                throw InvalidQRCodeException("Generated hash doesn't match QRCode hash")
```

Comment:
Please add a log statement when the hash matching fails.

Reply:
I guess this should be done globally as this is not the only exception that could be thrown. Doing it looks out of the scope for the current task.